File: _src/04/Chapter4.VB/Context.vb
Imports System.Data.Entity
Imports System.Data.Entity.ModelConfiguration.Conventions
Imports Chapter4.VB.Models
Imports Chapter4.VB.Maps

Public Class Context
    Inherits DbContext
    Public Sub New()
        MyBase.New("name=chapter4")
    End Sub
    Property People As DbSet(Of Person)
    Property Phones() As DbSet(Of Phone)
    Property PersonTypes() As DbSet(Of PersonType)
    Property Companies() As DbSet(Of Company)

    Protected Overrides Sub OnModelCreating(ByVal modelBuilder As DbModelBuilder)
        modelBuilder.Configurations.Add(New PersonMap)
        modelBuilder.Configurations.Add(New PersonTypeMap)
        modelBuilder.Conventions.Remove(Of OneToManyCascadeDeleteConvention)()
        modelBuilder.Conventions.Remove(Of ManyToManyCascadeDeleteConvention)()
    End Sub

End Class

File: _src/04/Chapter4.CSharp/Context.cs
using System.Data.Entity;
using System.Data.Entity.ModelConfiguration.Conventions;
using Chapter4.CSharp.Maps;
using Chapter4.CSharp.Models;

namespace Chapter4.CSharp
{
    public class Context : DbContext
    {
        public Context()
            : base("name=chapter4")
        {

        }
        public DbSet<Person> People { get; set; }
        public DbSet<Phone> Phones { get; set; }
        public DbSet<PersonType> PersonTypes { get; set; }
        public DbSet<Company> Companies { get; set; }
        protected override void OnModelCreating(DbModelBuilder modelBuilder)
        {
            modelBuilder.Configurations.Add(new PersonMap());
            modelBuilder.Configurations.Add(new PersonTypeMap());
            modelBuilder.Conventions.Remove<OneToManyCascadeDeleteConvention>();
            modelBuilder.Conventions.Remove<ManyToManyCascadeDeleteConvention>();
        }

    }
}
